lookout/style/format/analyzer.py

```diff
@@ -19,16 +19,25 @@
 from lookout.style.format.rules import TopDownGreedyBudget, TrainableRules
 
 
+def warmup_generator(func):
+    def start(*args, **kwargs):
+        cr = func(*args, **kwargs)
+        next(cr)
+        return cr
+    return start
+
+
 class FormatAnalyzer(Analyzer):
     log = logging.getLogger("FormatAnalyzer")
     model_type = FormatModel
     name = "style.format.analyzer.FormatAnalyzer"
     version = "1"
     description = "Source code formatting: whitespace, new lines, quotes, braces."
 
-    def __init__(self, model: AnalyzerModel, url: str, config: Mapping[str, Any]) -> None:
+    def __init__(self, model: AnalyzerModel, url: str, config: Mapping[str, Mapping[str, Any]]
+                 ) -> None:
         super().__init__(model, url, config)
-        self.config = self._load_config(self.config)
+        self._config_generator = self._load_config(self.config)
```

Comment:
I don't think a generator is necessary. Just merging the default config with the given one and putting it in self.config for later use in ChainMap should be enough.

Reply:
yes, you are right. I just think that in case we have a lot of languages no need to create all configs to analyze only one or two, which we have in the repo. Also, I do not want to make train function more complicated (even if it is just `ChainMap `). So I can create some object and make `config[language]` do all work in `train()`.
Is it ok?

Reply:
ok, just wanted to play with generators ;)